### ISSUE TITLE
Resize SVG before storing in memory

### ISSUE THREAD

**Issue by jjyoh** [lovell/sharp#2605]
I'm using Sharp as an "SVG to PNG converter" on AWS Lambda and if the SVG is crazy large (e.g. https://svgur.com/i/Uf3.svg), it returns a memory error :

`"Error: wbuffer_write: write failed\nunix error: No space left on device\nvips2png: unable to write to target target`

After changing the width & height of the SVG to a lower value, I was able to make it work. It seems that adding a feature for resizing an SVG before processing it could solve the issue. 

The workaround of adding a new env VIPS_DISC_THRESHOLD & increasing the memory of Lambda doesn't work for this one.

**Comment by lovell** [comment]
The `density` option can be used for this.
```js
sharp(input, { density: 20 })...
```

https://sharp.pixelplumbing.com/api-constructor

**Comment by jjyoh** [comment]
Yes, I'm actually already using the density param, but for my use case, I cannot reduce it to a number too small.

**Comment by lovell** [comment]
You could also try `sequentialRead: true`, which can reduce memory consumption:
```js
sharp(input, { density: 20, sequentialRead: true })...
```
